### PR TITLE
Concat validation set and training set for final_fit if using validation_split

### PR DESCRIPTION
**Pull request by jhfjhfj1**
### Which issue(s) does this Pull Request fix?

resolves #000

### CODE REVIEW

File: autokeras/auto_model.py [hunk 1]
```diff
@@ -139,6 +144,7 @@ def fit(self,
                     tf.keras.callbacks.EarlyStopping(patience=10)]
 
         self.tuner.search(x=dataset,
+                          concat=self._split_dataset,
```
Comment (fchollet) [inline autokeras/auto_model.py]: `search` is only called once per tuner instance, since the tuner is instantiated right above this line (L127). Would it make sense to have the argument be a Tuner constructor argument instead of being part of the `search` signature?

Also, `concat` as a name is not representative of what the argument means. Better to name it e.g. `fit_on_val_data`.
